data/doxygen.h

```diff
@@ -7,45 +7,83 @@
  * sequence of steps which are executed in order or through control flow statements. The
  * behavior of each step is defined in the LUA scripting language.
  *
- * The C++ classes for modeling sequences and steps are, unsurprisingly,
- * \ref task::Sequence "Sequence" and \ref task::Step "Step". A sequence acts like a
- * container for steps and can be executed in the current thread with the member function
+ * The library provides the main modeling classes for sequences and steps, functionality
+ * for executing them in the current thread or in a parallel one, as well as serialization
+ * support for saving and loading them.
+ *
+ * \section Creating a sequence and running it in the current thread
+ *
+ * The C++ classes for modeling sequences and steps are \ref task::Sequence "Sequence" and
+ * \ref task::Step "Step". A sequence acts like a container for steps and can be executed
+ * in the current thread with the member function
  * \ref task::Sequence::execute "Sequence::execute()":
  *
  * \code {.cpp}
  * #include <taskomat/taskomat.h>
  *
  * using namespace task;
  *
- * // Create some steps
- * Step step1{ Step::type_action };
- * step1.set_script("a = 42");
+ * Sequence create_sequence()
+ * {
+ *     // Create some steps
+ *     Step step1{ Step::type_action };
+ *     step1.set_script("a = 42");
+ *
+ *     Step step2{ Step::type_action };
+ *     step2.set_script("print('Mary had a little lamb.')");
+ *
+ *     // Assemble the steps into a sequence
+ *     Sequence sequence{ "An example sequence" };
+ *     sequence.push_back(step1);
+ *     sequence.push_back(step2);
  *
- * Step step2{ Step::type_action };
- * step2.set_script("b = 13");
+ *     return sequence;
+ * }
  *
- * // Assemble the steps into a sequence
- * Sequence sequence;
- * sequence.push_back(step1);
- * sequence.push_back(step2);
+ * Sequence sequence = create_sequence();
+ * Context context; // A context may contain additional information for the sequence
  *
  * // Run the sequence
- * sequence.execute();
+ * sequence.execute(context);
+ * \endcode
+ *
+ * \section Running a sequence in a parallel thread
+ *
+ * Taskolib can start a sequence in a parallel thread via an \ref task::Executor Executor
+ * object. The main thread should frequently call \ref task::Executor::update update() on
+ * this object to get information about the status of the execution. No other
+ * synchronization is necessary.
+ *
+ * \code {.cpp}
+ * #include <taskomat/taskomat.h>
+ *
+ * using namespace task;
+ *
+ * Executor ex;
+ * Sequence sequence = create_sequence(); // get a Sequence from somewhere
+ * Context context;
```

Comment:
Well, please ignore if I mentioned that already.
I usually add the constructor braces to make clear that here a new object is created.
But that might be my personal tic ;-D

```suggestion
 * Executor ex{ };
 * Sequence sequence = create_sequence();
 * Context context{ };
```

Reply:
I tend more towards seeing that as visual noise... :)